app/javascript/react/src/containers/UpvoteDownvote.js

```diff
@@ -0,0 +1,94 @@
+import React, { Component } from 'react'
+
+class UpvoteDownvote extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      currentUserVote: 0,
+      netVotes: 0
+    }
+    this.handleClick = this.handleClick.bind(this)
+  }
+
+  handleClick(event) {
+    let currentUserVote;
+    let netVotes = this.state.netVotes
+    if (event.target.className.split(" ")[1] === "fa-thumbs-down") {
+      if (this.state.currentUserVote === 0) {
+        currentUserVote = -1
+        netVotes += -1
+      } else if (this.state.currentUserVote === 1) {
+        currentUserVote = -1
+        netVotes += -2
+      } else if (this.state.currentUserVote === -1) {
+        currentUserVote = 0
+        netVotes += 1
+      }
+    } else if (event.target.className.split(" ")[1] === "fa-thumbs-up") {
+      if (this.state.currentUserVote === 0) {
+        currentUserVote = 1
+        netVotes += 1
+      } else if (this.state.currentUserVote === -1) {
+        currentUserVote = 1
+        netVotes += 2
+      } else if (this.state.currentUserVote === 1) {
+        currentUserVote = 0
+        netVotes += -1
+      }
+    }
```

Comment:
In the future, I would advise moving this logic into helper functions if you wish to impress employers